core/types/transaction_conditional.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2014 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/common/hexutility"
+)
+
+type KnownAccounts map[libcommon.Address]*Value
+
+type Value struct {
+	Single  *libcommon.Hash
+	Storage map[libcommon.Hash]libcommon.Hash
+}
+
+func SingleFromHex(hex string) *Value {
+	return &Value{Single: libcommon.HexToRefHash(hex)}
```

Comment:
```suggestion
	return &Value{Single: &libcommon.HexToHash(hex)}
```
with this there's no need for a new HexToRefHash function in erigon-lib